src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -26,17 +30,101 @@ export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurati
         token?: vscode.CancellationToken
     ): Promise<AwsSamDebuggerConfiguration | undefined> {
         if (!AWS_SAM_DEBUG_REQUEST_TYPES.has(debugConfiguration.request)) {
-            vscode.window.showInformationMessage('Invalid request type')
+            vscode.window.showErrorMessage('Invalid request type')
+
+            return undefined
+        }
+
+        if (!AWS_SAM_DEBUG_TARGET_TYPES.has(debugConfiguration.invokeTarget.target)) {
+            vscode.window.showErrorMessage('Invalid invokeTarget.target type')
+
+            return undefined
         }
-        if (
-            debugConfiguration.invokeTarget &&
-            debugConfiguration.invokeTarget.target &&
-            !AWS_SAM_DEBUG_TARGET_TYPES.has(debugConfiguration.invokeTarget.target)
-        ) {
-            vscode.window.showInformationMessage('Invalid invokeTarget.target type')
+
+        if (debugConfiguration.invokeTarget.target === 'template') {
+            if (!isTemplateDebugConfigValid(debugConfiguration)) {
+                return undefined
+            }
+        } else if (debugConfiguration.invokeTarget.target === 'code') {
+            if (
+                !debugConfiguration.lambda ||
+                !debugConfiguration.lambda.runtime ||
+                !samLambdaRuntimes.has(debugConfiguration.lambda.runtime)
+            ) {
+                vscode.window.showErrorMessage(
+                    'Debug config of invokeTarget.target: code must have a valid lambda.runtime'
```

Comment:
Suggestion: "Debug Configurations with an invoke target of 'code' require a valid Lambda runtime value."
(I'm not sure if we need to specify the exact field locations, that is an interesting one)